test/cli/cli_test.cpp

```diff
@@ -96,8 +121,8 @@ TEST_F(cli_test, with_argument_verbose)
 
 TEST_F(cli_test, with_out_file)
 {
-    cli_test_result result = execute_app("fastq_to_fasta", data("in.fastq"), "-o", data("out.fa"));
-    seqan3::sequence_file_input fin{data("out.fa"), seqan3::fields<seqan3::field::seq, seqan3::field::id>{}};
+    cli_test_result result = execute_app("fastq_to_fasta", data("in.fastq"), "-o", workdir("out.fasta"));
```

Comment:
```suggestion
    cli_test_result result = execute_app("fastq_to_fasta", data("in.fastq"), "-o", "out.fasta");
```

The app should be executed with the cwd = workdir.

Reply:
Good idea! So I will use `std::filesystem::current_path` to set the work directory... But this would result in every test creating a (possibly empty) output directory, even if no output files are produced. Is that fine?

Reply:
I think rene and I thought that a cwd will be created only temporary for the current executed test case and that it will be deleted afterwards. If someone wants to keep the files he should use an debugger with break points and copy out the files manually.

Reply:
I suggest we offer a target `make test_clean` that deletes the output directory. We want to motivate test-driven development, so the output must be easily available. Users who do not need the files can delete them with this simple make command. Is it a solution you can agree with?

Reply:
Not sure how cmake will know what it needs to delete, but you are right. We have to different scenarios someone who executes the test (does not want thousands of generated files that might be quite big) and the app-developer who wants to see the results when writing the test cases.

My gut feeling is that the tests will be executed way more (time-wise) in an automatic test (e.g. CI, does the app still build and produces my gold-standard results) setting than in a manual test where I care about the actual results. So I think that the default should be delete everything.

For example, cmake has an option

```
  --debug-trycompile           = Do not delete the try_compile build tree.
                                 Only useful on one try_compile at a time.
```

that keeps the intermediate files. Maybe we want something like this to when executing the test.

Something like
```terminal
> ./fastq2fasta_cli_test --keep-working-directories
```

Reply:
Thank you, that sounds like a good solution! I try implement this :)

Reply:
We should do the `--keep-working-directories` on a later PR, because we need to change the infrastructure to allow parsing new arguments.